src/Common/src/System/Net/Security/CertificateHelper.cs

```diff
@@ -0,0 +1,84 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Win32.SafeHandles;
+using System.Diagnostics;
+using System.Globalization;
+using System.Security.Cryptography.X509Certificates;
+
+namespace System.Net.Security
+{
+    internal static partial class CertificateHelper
+    {
+        private const string ClientAuthenticationOID = "1.3.6.1.5.5.7.3.2";
+
+        internal static X509Certificate2 GetEligibleClientCertificate()
+        {
+            // Get initial list of client certificates from the MY store.
+            X509Certificate2Collection candidateCerts;
+            using (var myStore = new X509Store(StoreName.My, StoreLocation.CurrentUser))
+            {
+                myStore.Open(OpenFlags.OpenExistingOnly | OpenFlags.ReadOnly);
+                candidateCerts = myStore.Certificates;
+            }
+            
+            return GetEligibleClientCertificate(candidateCerts);
+        }
+
+        internal static X509Certificate2 GetEligibleClientCertificate(X509CertificateCollection candidateCerts)
+        {
+            if (candidateCerts.Count == 0)
+            {
+                return null;
+            }
+
+            var certs = new X509Certificate2Collection();
+            certs.AddRange(candidateCerts);
```

Comment:
This code wasn't changed when I moved it.  But to answer your question, a new collection is being made because in this particular overload, the input parameter is a X509CertificateCollection.  And the collection is being "upgraded" to an X509Certificate2Collection in order to use the Find() methods on it when it then calls the other overload of GetElegibileClientCertificate().

The API surface from ClientWebSocket.Options only has a property that is a X509CertificateCollection.  Other code such as WinHttpHandler and HttpClientHandler (UAP) use only X509Certificate2Collection.  So, they don't call the overload that takes the X509CertificateCollection.